packages/taucmdr/cf/software/libelf_installation.py

```diff
@@ -45,9 +49,16 @@ class LibelfInstallation(AutotoolsInstallation):
     """Encapsulates a libelf installation."""
 
     def __init__(self, sources, target_arch, target_os, compilers):
+        if compilers[CC].unwrap().info.family is PGI or compilers[CC].unwrap().info.family is NVHPC :
+            try:
+                gnu_compilers = GNU.installation()
+            except ConfigurationError as err:
+                raise SoftwarePackageError("GNU compilers (required to build libelf) could not be found.") from err
+            compilers = compilers.modify(Host_CC=gnu_compilers[CC], Host_CXX=gnu_compilers[CXX])
         super().__init__('libelf', 'libelf', sources,
                                                  target_arch, target_os, compilers, REPOS, None, LIBRARIES, HEADERS)
 
     def configure(self, flags):
         flags.extend(['--disable-debuginfod'])
-        return super().configure(flags)
+        # Libelf's configure script requires CC to be a GNU compiler, so we have to set the environment variable before running the configure script
+        super().configure(flags, env={'CC':GNU.installation()[CC].unwrap().absolute_path})
```

Comment:
@vikram8128 very minor nit: could you return this last line?
```suggestion
        return super().configure(flags, env={'CC':GNU.installation()[CC].unwrap().absolute_path})
```